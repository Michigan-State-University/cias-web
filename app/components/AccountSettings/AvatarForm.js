import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import ConfirmationBox from 'components/ConfirmationBox';
import Row from 'components/Row';
import { themeColors, colors } from 'theme';

import messages from './messages';
import {
  StyledUserAvatar,
  StyledButtonsRow,
  StyledDeleteButton,
  StyledUploadFileButton,
} from './styled';

const AvatarForm = ({
  user: { avatar, firstName, lastName },
  addAvatar,
  deleteAvatar,
}) => {
  const [confirmationOpen, setConfirmationOpen] = useState(false);
  const openConfirmation = () => setConfirmationOpen(true);
  const closeConfirmation = () => setConfirmationOpen(false);

  const handleUpload = file => {
    addAvatar({
      image: file,
      imageUrl: window.URL.createObjectURL(file),
    });
  };

  const handleRemoval = () => {
    deleteAvatar();
    closeConfirmation();
  };

  return (
    <Fragment>
      <ConfirmationBox
        visible={confirmationOpen}
        onClose={closeConfirmation}
        description={<FormattedMessage {...messages.removeConfirmation} />}
        confirmAction={handleRemoval}
      />
      <Row align="center" width="100%">
        <StyledUserAvatar
          avatar={avatar}
          firstName={firstName}
          lastName={lastName}
        />
        <Row filled>
          <StyledButtonsRow>
            <StyledUploadFileButton
              textProps={{
                color: themeColors.secondary,
                fontWeight: 'bold',
                fontSize: 14,
              }}
              height="fit-content"
              onUpload={handleUpload}
            >
              <FormattedMessage
                {...messages[avatar ? 'changeAvatar' : 'addAvatar']}
              />
            </StyledUploadFileButton>
            {avatar && (
              <StyledDeleteButton
                color={colors.flamingo}
                fontWeight="bold"
                fontSize={14}
                onClick={openConfirmation}
              >
                <FormattedMessage {...messages.removeAvatar} />
              </StyledDeleteButton>
            )}
          </StyledButtonsRow>
        </Row>
      </Row>
    </Fragment>
  );
};

AvatarForm.propTypes = {
  user: PropTypes.shape({
    avatar: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
  }),
  addAvatar: PropTypes.func,
  deleteAvatar: PropTypes.func,
};

export default AvatarForm;
