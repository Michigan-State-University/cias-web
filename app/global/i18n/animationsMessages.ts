import { defineMessages } from 'react-intl';

import { NarratorAnimation } from 'models/Narrator';

export const scope = 'app.global.Animations';

export default defineMessages<NarratorAnimation>({
  uncertain: {
    id: `${scope}.uncertain`,
    defaultMessage: 'Uncertain',
  },
  wave: {
    id: `${scope}.wave`,
    defaultMessage: 'Wave',
  },
  greet: {
    id: `${scope}.greet`,
    defaultMessage: 'Greet',
  },
  otohLeftHand: {
    id: `${scope}.otohLeftHand`,
    defaultMessage: 'On the other hand (left)',
  },
  otohRightHand: {
    id: `${scope}.otohRightHand`,
    defaultMessage: 'On the other hand (right)',
  },
  pointRight: {
    id: `${scope}.pointRight`,
    defaultMessage: 'Point right',
  },
  pointLeft: {
    id: `${scope}.pointLeft`,
    defaultMessage: 'Point left',
  },
  pointUp: {
    id: `${scope}.pointUp`,
    defaultMessage: 'Point up',
  },
  pointDown: {
    id: `${scope}.pointDown`,
    defaultMessage: 'Point down',
  },
  pointLeftUp: {
    id: `${scope}.pointLeftUp`,
    defaultMessage: 'Point left up',
  },
  pointRightUp: {
    id: `${scope}.pointRightUp`,
    defaultMessage: 'Point right up',
  },
  explain: {
    id: `${scope}.explain`,
    defaultMessage: 'Explain',
  },
  rest: {
    id: `${scope}.rest`,
    defaultMessage: 'Rest',
  },
  restShoulderRubbing: {
    id: `${scope}.restShoulderRubbing`,
    defaultMessage: 'Rest shoulder rubbing',
  },
  restStretching: {
    id: `${scope}.restStretching`,
    defaultMessage: 'Rest stretching',
  },
  restWeightShift: {
    id: `${scope}.restWeightShift`,
    defaultMessage: 'Rest weight shift',
  },
  read: {
    id: `${scope}.read`,
    defaultMessage: 'Read',
  },
  reading: {
    id: `${scope}.reading`,
    defaultMessage: 'Reading',
  },
  confused: {
    id: `${scope}.confused`,
    defaultMessage: 'Confused',
  },
  doMagic: {
    id: `${scope}.doMagic`,
    defaultMessage: 'Do magic',
  },
  getAttention: {
    id: `${scope}.getAttention`,
    defaultMessage: 'Get attention',
  },
  search: {
    id: `${scope}.search`,
    defaultMessage: 'Search',
  },
  suggest: {
    id: `${scope}.suggest`,
    defaultMessage: 'Suggest',
  },
  surprised: {
    id: `${scope}.surprised`,
    defaultMessage: 'Surprised',
  },
  congratulate: {
    id: `${scope}.congratulate`,
    defaultMessage: 'Congratulate',
  },
  flyIn: {
    id: `${scope}.flyIn`,
    defaultMessage: 'Fly in',
  },
  flyOut: {
    id: `${scope}.flyOut`,
    defaultMessage: 'Fly out',
  },
  standStill: {
    id: `${scope}.standStill`,
    defaultMessage: 'Stand still',
  },
  process: {
    id: `${scope}.process`,
    defaultMessage: 'Process',
  },
  listen: {
    id: `${scope}.listen`,
    defaultMessage: 'Listen',
  },
  think: {
    id: `${scope}.think`,
    defaultMessage: 'Think',
  },
  write: {
    id: `${scope}.write`,
    defaultMessage: 'Write',
  },
  idle: {
    id: `${scope}.idle`,
    defaultMessage: 'Idle',
  },
  turnPage: {
    id: `${scope}.turnPage`,
    defaultMessage: 'Turn page',
  },
  openBook: {
    id: `${scope}.openBook`,
    defaultMessage: 'Open book',
  },
  closeBook: {
    id: `${scope}.closeBook`,
    defaultMessage: 'Close book',
  },
  announce: {
    id: `${scope}.announce`,
    defaultMessage: 'Announce',
  },
  blink: {
    id: `${scope}.blink`,
    defaultMessage: 'Blink',
  },
  acknowledge: {
    id: `${scope}.acknowledge`,
    defaultMessage: 'Acknowledge',
  },
  acknowledgeThoughtful: {
    id: `${scope}.acknowledgeThoughtful`,
    defaultMessage: 'Acknowledge thoughtful',
  },
  browsUp: {
    id: `${scope}.browsUp`,
    defaultMessage: 'Brows up',
  },
  decline: {
    id: `${scope}.decline`,
    defaultMessage: 'Decline',
  },
  declineAnnoyed: {
    id: `${scope}.declineAnnoyed`,
    defaultMessage: 'Decline annoyed',
  },
  declineThoughtful: {
    id: `${scope}.declineThoughtful`,
    defaultMessage: 'Decline thoughtful',
  },
  pleased: {
    id: `${scope}.pleased`,
    defaultMessage: 'Pleased',
  },
  sad: {
    id: `${scope}.sad`,
    defaultMessage: 'Sad',
  },
  hearLeftEar: {
    id: `${scope}.hearLeftEar`,
    defaultMessage: 'Hear left ear',
  },
  hearRightEar: {
    id: `${scope}.hearRightEar`,
    defaultMessage: 'Hear right ear',
  },
  hearBothEars: {
    id: `${scope}.hearBothEars`,
    defaultMessage: 'Hear both ears',
  },
  yawn: {
    id: `${scope}.yawn`,
    defaultMessage: 'Yawn',
  },
  eatCracker: {
    id: `${scope}.eatCracker`,
    defaultMessage: 'Eat cracker',
  },
  wearSunglasses: {
    id: `${scope}.wearSunglasses`,
    defaultMessage: 'Wear sunglasses',
  },
  lookDown: {
    id: `${scope}.lookDown`,
    defaultMessage: 'Look down',
  },
  lookDownAndBlink: {
    id: `${scope}.lookDownAndBlink`,
    defaultMessage: 'Look down and blink',
  },
  glanceUp: {
    id: `${scope}.glanceUp`,
    defaultMessage: 'Glance up',
  },
  glanceDown: {
    id: `${scope}.glanceDown`,
    defaultMessage: 'Glance down',
  },
  glanceLeft: {
    id: `${scope}.glanceLeft`,
    defaultMessage: 'Glance left',
  },
  glanceRight: {
    id: `${scope}.glanceRight`,
    defaultMessage: 'Glance right',
  },
  moveLeft: {
    id: `${scope}.moveLeft`,
    defaultMessage: 'Move left',
  },
  moveRight: {
    id: `${scope}.moveRight`,
    defaultMessage: 'Move right',
  },
});
