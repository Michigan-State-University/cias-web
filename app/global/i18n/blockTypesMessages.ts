import { defineMessages } from 'react-intl';

export const scope = 'app.global.BlockTypes';

export default defineMessages({
  BodyAnimation: {
    id: `${scope}.BodyAnimation`,
    defaultMessage: 'Body Animation',
  },
  Speech: {
    id: `${scope}.Speech`,
    defaultMessage: 'Speech',
  },
  Reflection: {
    id: `${scope}.Reflection`,
    defaultMessage: 'Speech',
  },
  ReflectionFormula: {
    id: `${scope}.ReflectionFormula`,
    defaultMessage: 'Speech',
  },
  HeadAnimation: {
    id: `${scope}.HeadAnimation`,
    defaultMessage: 'Head Animation',
  },
  ReadQuestion: {
    id: `${scope}.ReadQuestion`,
    defaultMessage: 'Read question',
  },
  Pause: {
    id: `${scope}.Pause`,
    defaultMessage: 'Pause',
  },
  Feedback: {
    id: `${scope}.Feedback`,
    defaultMessage: 'Show Spectrum',
  },
});
