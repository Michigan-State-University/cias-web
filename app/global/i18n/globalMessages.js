import { defineMessages } from 'react-intl';

export const scope = 'app.GlobalMessages';

export default defineMessages({
  variables: {
    variableNamePlaceholder: {
      id: `${scope}.variableNamePlaceholder`,
      defaultMessage: 'Variable name...',
    },
    variableScorePlaceholder: {
      id: `${scope}.variableScorePlaceholder`,
      defaultMessage: 'Score',
    },
    value: {
      id: `${scope}.value`,
      defaultMessage: 'Value:',
    },
    emptyBadge: {
      id: `${scope}.emptyBadge`,
      defaultMessage: 'No assigned value',
    },
  },
  blockTypes: {
    BodyAnimation: {
      id: `${scope}.BodyAnimation`,
      defaultMessage: 'Body Animation',
    },
    Speech: {
      id: `${scope}.Speech`,
      defaultMessage: 'Speech',
    },
    Reflection: {
      id: `${scope}.Reflection`,
      defaultMessage: 'Speech',
    },
    HeadAnimation: {
      id: `${scope}.HeadAnimation`,
      defaultMessage: 'Head Animation',
    },
    ReadQuestion: {
      id: `${scope}.ReadQuestion`,
      defaultMessage: 'Read question',
    },
  },
  questionTypes: {
    'Question::Single': {
      id: `${scope}.Question::Single`,
      defaultMessage: 'Single answer screen',
    },
    'Question::Multiple': {
      id: `${scope}.Question::Multiple`,
      defaultMessage: 'Multi answer screen',
    },
    'Question::TextBox': {
      id: `${scope}.Question::TextBox`,
      defaultMessage: 'Textbox screen',
    },
    'Question::Number': {
      id: `${scope}.Question::Number`,
      defaultMessage: 'Number screen',
    },
    'Question::Grid': {
      id: `${scope}.Question::Grid`,
      defaultMessage: 'Grid screen',
    },
    'Question::AnalogueScale': {
      id: `${scope}.Question::AnalogueScale`,
      defaultMessage: 'Visual Analogue Scale screen',
    },
    'Question::Information': {
      id: `${scope}.Question::Information`,
      defaultMessage: 'Information screen',
    },
    'Question::Url': {
      id: `${scope}.Question::Url`,
      defaultMessage: 'Url screen',
    },
    'Question::FollowUpContact': {
      id: `${scope}.Question::FollowUpContact`,
      defaultMessage: 'FollowUpContact screen',
    },
  },
  statuses: {
    draft: {
      id: `${scope}.draft`,
      defaultMessage: 'Draft',
    },
    published: {
      id: `${scope}.published`,
      defaultMessage: 'Published',
    },
    closed: {
      id: `${scope}.closed`,
      defaultMessage: 'Closed',
    },
    archived: {
      id: `${scope}.archived`,
      defaultMessage: 'Archived',
    },
  },
  createProblemError: {
    id: `${scope}.createProblemError`,
    defaultMessage: `Couldn't create an intervention`,
  },
  editProblemError: {
    id: `${scope}.editProblemError`,
    defaultMessage: `Couldn't edit an intervention`,
  },
});
