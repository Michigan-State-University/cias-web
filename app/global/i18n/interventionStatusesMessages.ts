import { defineMessages } from 'react-intl';

export const scope = 'app.global.InterventionStatusesMessages';

export default defineMessages({
  draft: {
    id: `${scope}.draft`,
    defaultMessage: 'Draft',
  },
  published: {
    id: `${scope}.published`,
    defaultMessage: 'Published',
  },
  closed: {
    id: `${scope}.closed`,
    defaultMessage: 'Closed',
  },
  archived: {
    id: `${scope}.archived`,
    defaultMessage: 'Archived',
  },
});
