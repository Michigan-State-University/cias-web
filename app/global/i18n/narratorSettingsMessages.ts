import { defineMessages } from 'react-intl';

import { NarratorSettingsKey } from 'models/Narrator';

export const scope = 'app.global.NarratorSettings';

export default defineMessages<NarratorSettingsKey>({
  [NarratorSettingsKey.VOICE]: {
    id: `${scope}.voice`,
    defaultMessage: 'Voice',
  },
  [NarratorSettingsKey.ANIMATION]: {
    id: `${scope}.animation`,
    defaultMessage: 'Display Narrator',
  },
  [NarratorSettingsKey.EXTRA_SPACE_FOR_NARRATOR]: {
    id: `${scope}.extra_space_for_narrator`,
    defaultMessage: 'Extra Space for Narrator',
  },
  [NarratorSettingsKey.CHARACTER]: {
    id: `${scope}.character`,
    defaultMessage: 'Character',
  },
});
