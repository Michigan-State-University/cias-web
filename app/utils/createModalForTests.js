import ReactDOM from 'react-dom';

const createModalForTests = () => {
  ReactDOM.createPortal = jest.fn(element => element);

  const modalContainer = document.createElement('div');
  modalContainer.setAttribute('id', 'modal-portal');
  document.body.appendChild(modalContainer);

  const mainAppContainer = document.createElement('div');
  mainAppContainer.setAttribute('id', 'main-app-container');
  document.body.appendChild(mainAppContainer);
};

export default createModalForTests;
