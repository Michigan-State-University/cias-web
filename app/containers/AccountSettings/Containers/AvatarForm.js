import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage, injectIntl } from 'react-intl';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import ConfirmationBox from 'components/ConfirmationBox';
import Row from 'components/Row';
import {
  makeSelectUser,
  addAvatarRequest,
  addAvatarSaga,
  deleteAvatarRequest,
  deleteAvatarSaga,
} from 'global/reducers/auth';
import { themeColors, colors } from 'theme';
import { useInjectSaga } from 'utils/injectSaga';

import messages from '../messages';
import {
  StyledUserAvatar,
  StyledButtonsRow,
  StyledDeleteButton,
  StyledUploadFileButton,
} from '../styled';

const AvatarForm = ({
  user: { avatar, firstName, lastName },
  addAvatar,
  deleteAvatar,
}) => {
  useInjectSaga({ key: 'addAvatar', saga: addAvatarSaga });
  useInjectSaga({ key: 'deleteAvatar', saga: deleteAvatarSaga });
  const [confirmationOpen, setConfirmationOpen] = useState(false);
  const openConfirmation = () => setConfirmationOpen(true);
  const closeConfirmation = () => setConfirmationOpen(false);

  const handleUpload = file => {
    addAvatar({
      image: file,
      imageUrl: window.URL.createObjectURL(file),
    });
  };

  const handleRemoval = () => {
    deleteAvatar();
    closeConfirmation();
  };

  return (
    <Fragment>
      <ConfirmationBox
        visible={confirmationOpen}
        onClose={closeConfirmation}
        description={<FormattedMessage {...messages.removeConfirmation} />}
        confirmAction={handleRemoval}
      />
      <Row align="center" width="100%">
        <StyledUserAvatar
          avatar={avatar}
          firstName={firstName}
          lastName={lastName}
        />
        <Row filled>
          <StyledButtonsRow>
            <StyledUploadFileButton
              textProps={{
                color: themeColors.secondary,
                fontWeight: 'bold',
                fontSize: 14,
              }}
              height="fit-content"
              onUpload={handleUpload}
            >
              <FormattedMessage
                {...messages[avatar ? 'changeAvatar' : 'addAvatar']}
              />
            </StyledUploadFileButton>
            {avatar && (
              <StyledDeleteButton
                color={colors.flamingo}
                fontWeight="bold"
                fontSize={14}
                onClick={openConfirmation}
              >
                <FormattedMessage {...messages.removeAvatar} />
              </StyledDeleteButton>
            )}
          </StyledButtonsRow>
        </Row>
      </Row>
    </Fragment>
  );
};

AvatarForm.propTypes = {
  user: PropTypes.shape({
    avatar: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
  }),
  addAvatar: PropTypes.func,
  deleteAvatar: PropTypes.func,
};

const mapStateToProps = createStructuredSelector({
  user: makeSelectUser(),
});

const mapDispatchToProps = {
  addAvatar: addAvatarRequest,
  deleteAvatar: deleteAvatarRequest,
};

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  injectIntl,
)(AvatarForm);
