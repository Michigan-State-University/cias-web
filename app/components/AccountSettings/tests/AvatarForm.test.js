/**
 *
 * Tests for AccountSettings
 *
 */

import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import 'jest-styled-components';
import { DEFAULT_LOCALE } from 'i18n';
import { IntlProvider } from 'react-intl';
import createModalForTests from 'utils/createModalForTests';

import AvatarForm from '../AvatarForm';

describe('<AvatarForm />', () => {
  beforeAll(() => {
    createModalForTests();
  });
  const defaultProps = {
    user: { avatar: 'src', firstName: 'test', lastName: 'test' },
    addAvatar: jest.fn(),
    deleteAvatar: jest.fn(),
  };

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <AvatarForm {...defaultProps} />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const { container } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <AvatarForm {...defaultProps} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should call remove avatar', () => {
    const { getByText } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <AvatarForm {...defaultProps} />
      </IntlProvider>,
    );
    const openModalButton = getByText('Remove profile photo');
    fireEvent.click(openModalButton);
    const removeAvatarButton = getByText('Confirm');
    fireEvent.click(removeAvatarButton);
    expect(defaultProps.deleteAvatar).toHaveBeenCalled();
  });
});
