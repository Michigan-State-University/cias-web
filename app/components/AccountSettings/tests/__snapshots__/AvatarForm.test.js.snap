// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvatarForm /> Should render and match the snapshot 1`] = `
.c6 {
  width: auto;
  height: auto;
  display: block;
  border-radius: 5px;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: auto;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1 {
  width: auto;
  height: auto;
  border-radius: 50%;
}

.c9 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  font-size: 14px;
  font-weight: 700;
  color: rgba(200,102,234,1);
  color: rgba(200,102,234,1);
}

.c10 {
  outline: none;
  border: none;
  background: transparent;
  font-weight: bold;
  font-size: 13px;
  line-height: 17px;
  padding: 2px;
  cursor: pointer;
  font-size: 14px;
  font-weight: 700;
  color: rgba(239,70,47,1);
}

.c2 {
  height: 75px;
  width: 75px;
  font-size: 40px;
}

.c11 {
  margin-left: 30px;
}

.c7 {
  margin-left: 20px;
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-left: 30px;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media only screen and (max-width:480px) {
  .c5 {
    margin-left: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    margin-top: 20px;
    margin-left: 20px;
  }
}

@media only screen and (max-width:360px) {
  .c11 {
    margin-left: 10px;
  }
}

@media only screen and (max-width:360px) {
  .c7 {
    margin-left: 10px;
  }
}

<div>
  <div
    class="c0"
    width="100%"
  >
    <img
      alt="avatar"
      class="c1 c2"
      src="src"
    />
    <div
      class="c3"
    >
      <div
        class="c4 c5"
      >
        <div
          class="c6 c7"
          height="fit-content"
        >
          <input
            autocomplete="off"
            style="display: none;"
            tabindex="-1"
            type="file"
          />
          <div
            class="c8"
          >
            <p
              class="c9"
              color="#C866EA"
              font-size="14"
              font-weight="bold"
            >
              Change profile photo
            </p>
          </div>
        </div>
        <button
          class="c10 c11"
          color="#EF462F"
          data-testid="open-modal"
          font-size="14"
          font-weight="bold"
        >
          Remove profile photo
        </button>
      </div>
    </div>
  </div>
</div>
`;
